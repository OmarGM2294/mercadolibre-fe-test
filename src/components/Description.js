import React from 'react'
import Image from 'next/image'

import styles from '../../styles/components/Description.module.scss'

const Description = (props) => (
  <main className={styles.description}>
    <div className={styles.container}>
      <Image
        className={styles.image}
        src={props.data.picture}
        height={680}
        width={680}
      />
      <div className={styles.info}>
        <span>
          {props.data.condition === 'new' ? 'Nuevo' : 'Usado'} - {props.data.sold_quantity} vendidos
        </span>
        <h1 className="h2">{ props.data.title }</h1>
        <h2 className="h1">
          $ { props.data.price.amount }
          <span>
            {(props.data.price.decimals + '0').slice(-2)}
          </span>
        </h2>
        <button className="button">
          Comprar
        </button>
      </div>
    </div>
    <div className={styles.productDescription}>
      <h3>Descripci√≥n del producto</h3>
      <p>
        {props.data.description}
      </p>
    </div>
  </main>
)

export default Description