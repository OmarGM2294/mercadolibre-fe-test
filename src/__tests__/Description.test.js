import React from 'react'

import { create } from 'react-test-renderer'

import Description from '../components/Description'

describe('Description', () => {
  it('Should match snapshot (it renders correctly)', () => {
    const component = create(<Description data={{
      id: 'MLA903779920',
      title: 'Vestido Holi- Modelo Johnny Cash- Ladrillo- Talle 1',
      price: { currency: 'ARS', amount: 5999, decimals: 10 },
      picture: '/imgs/search.svg',
      condition: 'new',
      free_shipping: true,
      sold_quantity: 0,
      description: 'Vestido Holi\nModelo Johnny Cash\nColor ladrillo\nTalle 1'
    }}/>)

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
