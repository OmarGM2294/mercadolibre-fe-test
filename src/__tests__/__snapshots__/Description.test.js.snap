// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description Should match snapshot (it renders correctly) 1`] = `
<main>
  <div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjgwIiBoZWlnaHQ9IjY4MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <noscript>
        <img
          decoding="async"
          src="/imgs/search.svg"
          srcSet="/_next/image?url=%2Fimgs%2Fsearch.svg&w=750&q=75 1x, /_next/image?url=%2Fimgs%2Fsearch.svg&w=1920&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
      <img
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </div>
    <div>
      <span>
        Nuevo
         - 
        0
         vendidos
      </span>
      <h1
        className="h2"
      >
        Vestido Holi- Modelo Johnny Cash- Ladrillo- Talle 1
      </h1>
      <h2
        className="h1"
      >
        $ 
        5999
        <span>
          00
        </span>
      </h2>
      <button
        className="button"
      >
        Comprar
      </button>
    </div>
  </div>
  <div>
    <h3>
      Descripci√≥n del producto
    </h3>
    <p>
      Vestido Holi
Modelo Johnny Cash
Color ladrillo
Talle 1
    </p>
  </div>
</main>
`;
